client/blocks/stripe-utils/utils.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * External dependencies
+ */
+import { getSetting } from '@woocommerce/settings';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { normalizeLineItems } from './normalize';
+import { errorTypes, errorCodes } from './constants';
+
+/**
+ * @typedef {import('./type-defs').StripeServerData} StripeServerData
+ * @typedef {import('./type-defs').StripePaymentItem} StripePaymentItem
+ * @typedef {import('./type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').PreparedCartTotalItem} CartTotalItem
+ */
+
+/**
+ * Stripe data comes form the server passed on a global object.
+ *
+ * @return  {StripeServerData} Stripe server data.
+ */
+const getStripeServerData = () => {
+	const stripeServerData = getSetting( 'stripe_data', null );
+	if ( ! stripeServerData ) {
+		throw new Error( 'Stripe initialization data is not available' );
+	}
+	return stripeServerData;
+};
```

Comment:
I just realized while reviewing something else, that there's a lot of places this function is called without checking whether a property on the returned object actually exists. While this function does throw an error if `getSetting( 'stripe_data', null )` returns a non truthy value, it does not do any validation of the actual properties on the data object. 

There's a couple ways of handling this:

- review everywhere this function is called and make sure the implemented properties are checked before using.
- do some validation of the properties and make sure that this function always returns a consistent shape (and potentially filling in any missing property values with relevant/correct defaults).

The latter is probably the better way to go (and would make for easier type-checking in later implementations of TypeScript).